test/meta/src/RuntimeTypes.c

```diff
@@ -262,6 +262,35 @@ void RuntimeTypes_ctor(void) {
     ecs_fini(world);
 }
 
+/* Tests that an illegal constructor is set for a struct if at least a member has
+ * itself an illegal constructor */
+void RuntimeTypes_ctor_illegal(void) {
+    ecs_world_t *world = ecs_init();
+
+    /* Define NestedStruct: */
+    const ecs_type_info_t *nested_struct_ti = define_nested_struct(world);
+
+    ecs_type_hooks_t hooks = nested_struct_ti->hooks;
+    hooks.flags |= ECS_CTOR_ILLEGAL; /* mark constructor for "NestedStruct" as illegal */
+    ecs_set_hooks_id(world, nested_struct, &hooks);
```

Comment:
This functionality should be tested in isolation (in a future PR) since it's part of the core, in `test/core/ComponentLifecycle`

Reply:
Makes sense. I would leave it in `RuntimeTypes.c` too, because in this case is testing the illegal flag propagates to derivative types.

Reply:
Agreed, these tests should remain in RuntimeTypes. What I meant was that the illegal flag feature by itself (e.g. setting it results in a callback that asserts) should be tested in isolation, as part of the core.

I'm about to merge a bunch of core tests for this.